File: contents/euclidean_algorithm/code/php/euclidean.php
<?php
declare(strict_types=1);

function euclid_sub(float $a, float $b): float
{
  $a = abs($a);
  $b = abs($b);

  while ($a !== $b) {
    if ($a > $b) $a = $a - $b;
    else $b = $b - $a;
  }

  return $a;
}

function euclid_mod(float $a, float $b): float
{
  $a = abs($a);
  $b = abs($b);

  while ($b != 0)
    list($b, $a) = [$a % $b, $b];

  return $a;
}

echo sprintf('Euclidean mod: %s', euclid_mod(64 * 67, 64 * 81));
echo PHP_EOL;
echo sprintf('Euclidean sub: %s', euclid_sub(128 * 12, 128 * 77));

File: contents/euclidean_algorithm/code/cpp/euclidean.cpp
#include <cmath>
#include <iostream>
#include <utility>

// Euclidean algorithm using modulus
int euclid_mod(int a, int b) {
  a = std::abs(a);
  b = std::abs(b);

  while (b != 0) {
    a = std::exchange(b, a % b);
  }

  return a;
}

// Euclidean algorithm with subtraction
int euclid_sub(int a, int b) {
  a = std::abs(a);
  b = std::abs(b);

  while (a != b) {
    if (a > b) {
      a -= b;
    } else {
      b -= a;
    }
  }

  return a;
}

int main() {
  auto check1 = euclid_mod(64 * 67, 64 * 81);
  auto check2 = euclid_sub(128 * 12, 128 * 77);

  std::cout << check1 << '\n';
  std::cout << check2 << '\n';
}
